packages/frontend/web/src/components/cocktail-detail/Carousel.tsx

```diff
@@ -0,0 +1,111 @@
+import { ChevronLeft, ChevronRight } from 'lucide-react';
+import { useState } from 'react';
+import { Link } from 'react-router-dom';
+
+import type { Cocktail } from '@app/types';
+
+const modulo = (number: number, modulus: number) => {
+  return ((number % modulus) + modulus) % modulus;
+};
+
+function getClassName(itemIndex: number) {
+  switch (itemIndex) {
+    case 0: {
+      return 'card border-dark absolute bg-card-green h-[336px] w-[288px] rounded-sm border-[3px]';
+    }
+    case 1: {
+      return 'border-dark bg-card-pink absolute h-[336px] w-[288px] rotate-6 rounded-sm border-[3px]';
+    }
+    case 2: {
+      return 'border-dark bg-pastel-yellow absolute h-[336px] w-[288px] -rotate-6 rounded-sm border-[3px]';
+    }
+    default: {
+      return 'hidden';
+    }
+  }
+}
+
+export default function Carousel({
+  similarCocktails,
+}: {
+  readonly similarCocktails: Cocktail[];
+}) {
+  const [index, setIndex] = useState(0);
+
+  const handlePrevious = () => {
+    setIndex((currentIndex) =>
+      modulo(currentIndex - 1, similarCocktails.length),
```

Comment:
```suggestion
      (currentIndex - 1) % similarCocktails.length,
```
Juste ça, ça suffit `number % modulus`